template/template.js

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const {promisify} = require('util');
+const recursiveCopy = promisify(require('recursive-copy'));
+
+const mkdir = promisify(fs.mkdir);
+const readdir = promisify(fs.readdir);
+const readFile = promisify(fs.readFile);
+const stat = promisify(fs.stat);
+const writeFile = promisify(fs.writeFile);
+
+module.exports = async templatePath => {
+  const contents = await readFile(templatePath, {encoding: 'UTF-8'});
+
+  const map = {};
+  const parts = contents.split(/<!-- (DOCS \w+) -->/).map((part, index) => {
```

Comment:
Just calling out that this would only split on the first instance of this pattern. `/g` needed if you intend to split on all occurrences.

Reply:
It seems to fully split the string.